BrainPortal/app/views/custom_filters/_userfile_custom_filter.html.erb

```diff
@@ -22,70 +22,100 @@
-<%= text_field_tag "data[size_term]", custom_filter.data["size_term"] %> kb
-
-<p>
-Owner<br>
-<%= user_select("data[user_id]", { :selector => custom_filter.data["user_id"] }, { :include_blank => true })  %>
-
-<p>
-Project<br>
-<%= group_select("data[group_id]", { :selector => custom_filter.data["group_id"]}, { :include_blank => true })  %>
-
-<p>
-  Data Provider<br>
-  <%= data_provider_select("data[data_provider_id]", { :selector => custom_filter.data["data_provider_id"] || '!' }, { :include_blank => true })  %>
-</p>
-
-<p>
-Archiving status<br>
-<%= select :data, :archiving_status,
-[
-  ["Don't filter by archiving status", ''        ],
-  ["Archived",                         'archived' ],
-  ["Not archived",                     'none'    ],
-], :selected => custom_filter.data["archiving_status"] %>
-
-<p>
-  Synchronization status<br>
-  <%=
-  select_tag "data[sync_status]", options_for_select(["InSync","ProvNewer","CacheNewer","Corrupted","ToCache","ToProvider"], :selected => custom_filter.data["sync_status"]),
-              :multiple => true, :size => 6, :selected => custom_filter.data["sync_status"]
+
+<%= show_table(@custom_filter, :edit_condition => (check_role(:admin_user) || @custom_filter.user_id == current_user.id)) do |t| %>
+
+  <% t.edit_cell(:name) do %>
+    <%= text_field_tag "custom_filter[name]", @custom_filter.name %>
+  <% end %>
+
+  <%
+    file_name_type =
+      {
+        ''        => "Don't filter name",
+        'match'   => 'Matches',
+        'contain' => 'Contains',
+        'begin'   => 'Begins with',
+        'end'     => 'Ends with',
+      }
   %>
-</p>
 
-<% if current_user.available_tags.count > 0 %>
-  Tags <br>
-  <%=
-  select_tag "data[tag_ids]", options_for_select((current_user.available_tags.map { |tags| [tags.name, tags.id] }), :selected => custom_filter.data["tag_ids"]),
-              :multiple => true, :size => 6, :selected => custom_filter.data["tag_ids"]
+  <% t.edit_cell(:data_file_name_type, :header => "Filename",
+                 :content => "#{file_name_type[@custom_filter.data_file_name_type]} #{@custom_filter.data_file_name_term}") do %>
+    <%= select :data, :file_name_type, file_name_type.map{|k,v| [v,k]}, :selected => @custom_filter.data_file_name_type %> <%= text_field_tag "data[file_name_term]", @custom_filter.data["file_name_term"] %>
+  <% end %>
+
+   <% t.edit_cell(:data_parent_name_like, :header => "Parent name contains (lists children)") do %>
+    <%= text_field_tag "data[parent_name_like]", @custom_filter.data[:child_name_like] %>
+  <% end %>
+
+  <% t.edit_cell(:data_child_name_like, :header => "Child name contains (lists parents)") do %>
+    <%= text_field_tag "data[child_name_like]", @custom_filter.data[:parent_name_like] %>
+  <% end %>
+
+  <% t.edit_cell(:data_type, :header => "By file type", :content => (@custom_filter.data_type||[]).sort.join(", ")) do %>
```

Comment:
Note: I needed to introduce a `.presence` after the getter method call to support cases with empty strings in the the data structure.